docs/source/federation-spec.md

```diff
@@ -413,3 +413,25 @@ type User @key(fields: "id") {
   name: String @override(from: "SubgraphA")
 }
 ```
+### `@inaccessible`
+```graphql
+directive @inaccessible(from: String!) on FIELD_DEFINITION | INTERFACE | OBJECT | UNION | ARGUMENT_DEFINITION | SCALAR | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION
+```
+
+The `@inaccessible` directive is used to indicate that a location within the schema is inaccessible. Inaccessible types and fields are available internally but are not exposed publically. The @inaccessible directive is helpful for adding a new field to a shared value type. Often when you add a value type field in one subgraph, composition fails because that field isn't resolvable in other subgraphs. This directive enables you to preserve composition while adding the field to your remaining subgraphs. When the rollout is complete, you can remove the @inaccessible directive and begin using the field.
+
+```graphql
+extend schema
```

Comment:
@jeffjakub this is the supergraph usage of `@link`.

the [subgraph usage of `@inaccessible` uses `import` from the federation/v2.0 spec](https://github.com/apollographql/supergraph-demo-fed2/blob/0876cd37626c3f582054386901d983ecf2d87777/subgraphs/products/products.graphql#L2).